From 2b4d8616f581a36ed98a491ac1ec14be69d37511 Mon Sep 17 00:00:00 2001
From: Jesse Barnes <jbarnes@virtuousgeek.org>
Date: Thu, 1 Apr 2010 17:26:04 -0700
Subject: [PATCH 01/11] GLX/DRI2: pass GLX drawable ID to dri2InvalidateBuffers

The IDs will be the same in the case where an X window is used directly
as a GLX drawable, but will fail if a new GLX drawable is created
explicitly, as with glxgears_fbconfig.

Fixes fdo bug #27190.
---
 src/glx/dri2_glx.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git Mesa-7.8/src/glx/dri2_glx.c Mesa-7.8/src/glx/dri2_glx.c
index 14788b8..c4b5cb9 100644
--- Mesa-7.8/src/glx/dri2_glx.c
+++ Mesa-7.8/src/glx/dri2_glx.c
@@ -329,7 +329,7 @@ dri2FlushFrontBuffer(__DRIdrawable *driDrawable, void *loaderPrivate)
 
    /* Old servers don't send invalidate events */
    if (!pdp->invalidateAvailable)
-       dri2InvalidateBuffers(priv->dpy, pdraw->base.xDrawable);
+       dri2InvalidateBuffers(priv->dpy, pdraw->base.drawable);
 
    dri2WaitGL(loaderPrivate);
 }
@@ -393,7 +393,7 @@ dri2SwapBuffers(__GLXDRIdrawable *pdraw, int64_t target_msc, int64_t divisor,
 
     /* Old servers don't send invalidate events */
     if (!pdp->invalidateAvailable)
-       dri2InvalidateBuffers(dpyPriv->dpy, pdraw->xDrawable);
+       dri2InvalidateBuffers(dpyPriv->dpy, pdraw->drawable);
 
     /* Old servers can't handle swapbuffers */
     if (!pdp->swapAvailable) {
-- 
1.7.0.3

